deltacat/storage/model/metafile.py

```diff
@@ -371,8 +380,11 @@ def path(self) -> Optional[str]:
             else None
         )
 
+    def is_undefined(self) -> bool:
+        return self.get("undefined", False)
```

Comment:
Instead of introducing a new "undefined" parameter that is stored/maintained separately, what do you think about just tying this back to `return bool(self.revision)` to ensure that we enforce valid/defined revision numbers as any non-zero value?